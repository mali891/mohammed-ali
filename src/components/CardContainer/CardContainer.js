import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { styles } from './CardContainer.styles'

const CardContainer = ({ children, marginBottom, className, ...other }) => {
  const dynamicClassNames = classNames([className], {
    'c-card-container': true,
    'c-card-container--margin-bottom': marginBottom
  })

  return (
    <div className={dynamicClassNames} css={styles} {...other}>
      {children}
    </div>
  )
}

CardContainer.defaultProps = {
  marginBottom: false
}

CardContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  marginBottom: PropTypes.bool
}

export default CardContainer
