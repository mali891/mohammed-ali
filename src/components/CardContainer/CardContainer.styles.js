import { css } from '@emotion/core'

import { MEDIA_QUERIES, SPACING } from '../../styles/settings'

export const styles = css`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  @media (min-width: ${MEDIA_QUERIES.DESKTOP}) {
    &.c-card-container--margin-bottom {
      margin-bottom: ${SPACING.DESKTOP.SM};
    }
  }
`
