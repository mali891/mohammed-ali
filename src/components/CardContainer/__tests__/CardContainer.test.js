import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import CardContainer from '../CardContainer'

describe('CardContainer', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<CardContainer />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given a marginBottom prop', () => {
    const { asFragment, container } = render(<CardContainer marginBottom />)

    expect(container.querySelector('.c-card-container--margin-bottom')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
