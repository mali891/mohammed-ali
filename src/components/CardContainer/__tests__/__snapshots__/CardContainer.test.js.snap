// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-card-container"
    css="[object Object]"
  />
</DocumentFragment>
`;

exports[`CardContainer renders correctly given a marginBottom prop 1`] = `
<DocumentFragment>
  <div
    class="c-card-container c-card-container--margin-bottom"
    css="[object Object]"
  />
</DocumentFragment>
`;
